src/endpoint/s3/s3_rest.js

```diff
@@ -287,14 +287,14 @@ async function authorize_request_policy(req) {
     // we start the permission check on account identifier intentionally
     if (account_identifier_id) {
         permission_by_id = await s3_bucket_policy_utils.has_bucket_policy_permission(
-            s3_policy, account_identifier_id, method, arn_path, req);
+            s3_policy, {account_identifier_name, account_identifier_id}, method, arn_path, req);
```

Comment:
The account argument is a ID/name (string), not an object.
Continue with the flow, and you will see in `_is_principal_fit` the check of `principal.unwrap() === account`. You can add logs to better understand where you are (before the fix).

This check is only in NC and it checks the principal by ID (see "if (account_identifier_id) {").

Reply:
@shirady any reason why we are passing id instead of name ? However the account name will be unique and we can use the name instead of id

Reply:
It is an additional way that we allow to add Principals, you can see in the [S3Ops.md](https://github.com/noobaa/noobaa-core/blob/master/docs/NooBaaNonContainerized/S3Ops.md#principal-field), the reason was to have it like in AWS for accounts, and the initial thought that was that t is part of solving the issue about account names - see issue #7734

Reply:
So we should have account id work for NC as well rather than going for Account name ?

Reply:
No, we support both options. We support the account name for backward compatibility.